src/repositories/video/video.repository.ts

```diff
@@ -2,56 +2,36 @@ import { Model } from 'mongoose';
 import { Injectable } from '@nestjs/common';
 import { InjectModel } from '@nestjs/mongoose';
 import { Video } from './schemas/video.schema';
-import { DbVideoToPublishDto, dbVideoToPublishProjection } from './dto/videos-to-publish.dto';
 import { UpdateResult } from 'mongodb';
 import moment from 'moment';
 
-type TrendingChainResponse = Array<{
-  permlink: string;
-  title: string;
-  duration: number;
-  created: Date;
-  language: string;
-  views: number;
-  trending_position: number;
-  author: string;
-  images: {
-    thumbnail: string;
-    poster: string;
-    post: string;
-  };
-  tags: string[];
-}>;
-
 @Injectable()
 export class VideoRepository {
   constructor(@InjectModel(Video.name, 'threespeak') private videoModel: Model<Video>) {}
 
-  async getVideosToPublish(): Promise<DbVideoToPublishDto[]> {
+  async getVideosToPublish(): Promise<Video[]> {
     return await this.videoModel
-      .find(
-        {
-          status: 'published',
-          publishFailed: { $ne: true },
-          lowRc: { $ne: true },
-          owner: { $ne: 'guest-account' },
-          $or: [{ steemPosted: { $exists: false } }, { steemPosted: false }],
-          title: { $ne: null },
-        },
-        dbVideoToPublishProjection,
```

Comment:
Did the projection break something? Or is it just legacy?

Reply:
The projection was based on a DTO, which was problematic because it meant there were two sources of truth for the shape of the data—the DTO and the Mongoose schema. For the minor efficiency gain, it's really not worth it. Custom projections should be used where necessary.

Reply:
got it